hhds/bench/scripts/plot.py

```
@@ -0,0 +1,136 @@
+
+    # Color palette: matplotlib default cycles 10 colors; for >10 ops we
+    # rely on tab20.
+    cmap = plt.get_cmap("tab20" if len(ops) > 10 else "tab10")
+    op_color = {op: cmap(i % cmap.N) for i, op in enumerate(ops)}
+
+    # Markers distinguish the OTHER axes (library and/or topology) so a
+    # single op's color appears as a family of related lines.
+    markers = ["o", "s", "^", "D", "v", "P", "X", "*"]
+    # Pair each (library, topology) combo with one marker.
+    lib_topo_marker = {}
+    for i, (lib, topo) in enumerate(sorted({(l, t) for l in libs for t in topos})):
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)